### PR TITLE
Replace all pagination with Filter classes

### PR DESCRIPTION
**Pull request by giggsey**
Continuation of #259

### CODE REVIEW

**Comment by giggsey** [comment]
I force pushed to try to get rid of some of the unrelated commits, but I think the pipeline is from before. CS Fixer runs fine locally

File: src/Controller/OrganizationController.php [hunk 1]
```diff
@@ -24,12 +24,12 @@
 use Buddy\Repman\Message\Organization\RemoveToken;
 use Buddy\Repman\Message\Organization\SynchronizePackage;
 use Buddy\Repman\Message\Security\ScanPackage;
-use Buddy\Repman\Query\Filter;
 use Buddy\Repman\Query\User\Model\Installs\Day;
 use Buddy\Repman\Query\User\Model\Organization;
 use Buddy\Repman\Query\User\Model\Package;
 use Buddy\Repman\Query\User\OrganizationQuery;
 use Buddy\Repman\Query\User\PackageQuery;
+use Buddy\Repman\Query\User\PackageQuery\Filter;
```
Comment (akondas) [inline src/Controller/OrganizationController.php:32]:
```suggestion
use Buddy\Repman\Query\User\PackageQuery\Filter as PackageFilter;
```

File: src/Query/User/PackageQuery.php [hunk 2]
```diff
@@ -4,13 +4,13 @@
 
 namespace Buddy\Repman\Query\User;
 
-use Buddy\Repman\Query\Filter;
```
Comment (akondas) [inline src/Query/User/PackageQuery.php:7]: same as above, use alias :wink:

File: src/Query/User/PackageQuery.php [hunk 3]
```diff
@@ -22,6 +22,11 @@ public function findAll(string $organizationId, Filter $filter): array;
 
     public function count(string $organizationId, Filter $filter): int;
 
+    /**
+     * @return Package[]
+     */
+    public function find(string $organizationId, string $searchString, int $limit = 20, int $offset = 0): array;
```
Comment (akondas) [inline src/Query/User/PackageQuery.php:28]: :thinking: looks like this isn't used anywhere

File: src/Query/User/PackageQuery/Filter.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    public function hasSearchTerm(): bool
+    {
+        return $this->searchTerm !== null;
+    }
+
+    public function getQueryStringParams(): array
+    {
+        $params = \Buddy\Repman\Query\Filter::getQueryStringParams();
```
Comment (akondas) [inline src/Query/User/PackageQuery/Filter.php:35]: please, try always to avoid FQCN

File: src/Query/User/PackageQuery/Filter.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @author giggsey
+ */
```
Comment (akondas) [inline src/Query/User/PackageQuery/Filter.php:6]: I suspect that it is inserted from template :wink:

**Review by akondas — COMMENTED** [review]
minor touches to make, I improved the code style but now phpstan is still complaining about something, is the status of this build visible to you?

**Comment by giggsey** [comment]
Both PHPStan failures are because `TokenFixtures` and `PrivatePackageDownloadFixtures` want to load 9999 organisations. From what I can see, both classes are only used to populate dummy data. Happy to them to limit to 100 instead?

**Review by akondas — APPROVED** [review]
Thanks :beers: :cocktail:

File: src/Query/User/PackageQuery/DbalPackageQuery.php [hunk 6]
```diff
@@ -144,7 +143,7 @@ public function versionCount(string $packageId): int
     /**
      * @return Version[]
      */
-    public function getVersions(string $packageId, int $limit = 20, int $offset = 0): array
+    public function getVersions(string $packageId, \Buddy\Repman\Query\Filter $filter): array
```
Comment (akondas) [inline src/Query/User/PackageQuery/DbalPackageQuery.php:146]: I wonder if we can somehow automate the detection of this notation?

Comment (giggsey) [reply src/Query/User/PackageQuery/DbalPackageQuery.php]: Possibly 'fully_qualified_strict_types', but it looks like that only works when the use statement is there in the first place

**Comment by akondas** [comment]
Thanks @giggsey for your work :beers:

**Pull request merged**